python-threatexchange/threatexchange/signal_type/index.py

```diff
@@ -139,6 +140,12 @@ def __eq__(self, other: t.Any) -> bool:
 IndexMatch = IndexMatchUntyped[SignalSimilarityInfo, T]
 
 
+@dataclass
+class IndexMatchWithRotation(t.Generic[T]):
+    match: IndexMatchUntyped[SignalSimilarityInfo, T]
+    rotation_type: RotationType = RotationType.ORIGINAL
```

Comment:
When you move it into the match command, consider making this optional, then adding a `__str__` command which handles the output of the match command. If rotation_type is None, then no rotation is printed. If the rotation is there, then it can format the string you put to string.